sqlx-core/src/logging.rs

```diff
@@ -0,0 +1,35 @@
+/// Logs the query and execution time of a statement as it runs.
+macro_rules! log_execution {
+    ( $query:expr, $block:expr ) => {{
+        // TODO: Log bound parameters
+        let query_string = $query.query_string();
+        let timer = std::time::Instant::now();
+        let result = $block;
+        let elapsed = timer.elapsed();
+        if elapsed >= std::time::Duration::from_secs(1) {
+            log::warn!(
+                "{} ..., elapsed: {:.3}s\n\n    {}\n",
+                crate::logging::parse_query_summary(query_string),
+                elapsed.as_secs_f32(),
+                query_string
+            );
+        } else {
+            log::debug!(
+                "{} ..., elapsed: {:.3}ms\n\n    {}\n",
```

Comment:
`Duration` has a `Debug` representation where it does the "nearest fit" and will show `45 ns` or `12 s` as needed

Reply:
TIL!